tmt/steps/prepare/multihost.py

```diff
@@ -0,0 +1,54 @@
+import tmt.utils
+
+
+class PrepareMultihost(tmt.steps.prepare.PreparePlugin):
+    """
+    Prepare the guest for running a multihost test.
+
+    This step is enabled implicitly, when multiple guests are detected in
+    the plan. It exports the information about guest roles and updates
+    /etc/hosts accordingly. Default order is '65'.
+
+    This method requires specifying roles and hosts. The expected format is
+    the following:
+
+    roles:
+      - server:
+          - server-one
+          - server-two
+
+    hosts:
+      - server-one: 10.10.10.10
+      - server-two: 10.10.10.11
+
+    The exported roles are comma-separated.
+    """
+
+    # Supported methods
+    _methods = [tmt.steps.Method(
+        name='multihost',
+        doc=__doc__,
+        order=tmt.utils.DEFAULT_PLUGIN_ORDER_MULTIHOST)]
+
+    # Supported keys
+    _keys = ['roles', 'hosts']
+
+    def default(self, option, default=None):
+        """ Return default data for given option """
+        if option in ('roles', 'hosts'):
+            return {}
+        return default
+
+    def go(self, guest):
+        """ Prepare the guests """
+        super().go(guest)
+
+        self.debug('Export roles.', level=2)
+        for role, corresponding_guests in self.get('roles').items():
+            formatted_guests = ','.join(corresponding_guests)
+            self.step.plan.environment[role] = formatted_guests
```

Comment:
This doesn't work and I can't find out why - check 'environment' in debug log - role is never set there.
I've added print on next line and self.step.plan.environment has only TMT_PLAN_DATA set

Reply:
Also I don't see any quoting - spaces in role name might make it difficult

Reply:
Maybe because environment is a property

Reply:
It works with `self.step.plan._environment[role] = formatted_guests`

Reply:
Yes, `Plan.environment` is a property. It would be better to handle this a bit more nicely to store the roles. What about storing roles separately and aggregating them as part of the property handling?

Reply:
For now let's just update `Plan._environment`, we can polish later.